### PR TITLE
Add `flake8-eradicate` to flake8 `pre-commit` hook

### PR DESCRIPTION
**Pull request by flpStrri**
Add `flake8-eradicate` `pre-commit` hook to protect committing commented code.

## Description

- Added the `additional_dependency` to `flake8`;
- Also added to `pyproject.toml`, even if not needed, so we can manage version updates;

## Types of Changes

- [ ] Core
- [x] Tooling
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Documentation

## Issues Fixed or Closed by This PR

#430

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the CONTRIBUTING document.
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by flpStrri** [comment]
It stopped me from committing on a file with commented code: Captura de Tela 2020-09-04 às 08 17 43

**Comment by patrick91** [comment]
Very nice! thank you for this 😊

I'll send a commit later today to remove or uncomment the code, if that's ok 😊

**Comment by flpStrri** [comment]
I think this PR is blocked by this issue: https://github.com/myint/eradicate/issues/24

**Comment by botberry** [comment]
Thanks for adding the `RELEASE.md` file!



Here's a preview of the changelog:

---

Improves tooling by adding `flake8-eradicate` to `flake8` `pre-commit` hook..

File: tests/conftest.py [hunk 1]
```diff
@@ -1,25 +1,2 @@
...
-# @strawberry.type
-# class Query:
-#     @strawberry.field
-#     def user(self, info) -> User:
-#         return User(name="Patrick", age=100)
-
-
-# @pytest.fixture
-# def schema():
-#     return strawberry.Schema(query=Query)
```
Comment (patrick91) [inline tests/conftest.py:21]: 👍

File: tests/federation/test_printer.py [hunk 2]
```diff
@@ -66,20 +66,3 @@ def top_products(self, info, first: int) -> List[Product]:
     assert schema.as_str() == textwrap.dedent(expected).strip()
 
     del Review
-
-
-# type Review {
```
Comment (patrick91) [inline tests/federation/test_printer.py:71]: 👍

File: tests/schema/test_generics.py [hunk 3]
```diff
@@ -349,46 +349,6 @@ def example(self, info) -> typing.Union[Fallback, Edge[int]]:
     }
 
 
-# def test_supports_generic_in_unions_multiple_vars():
```
Comment (patrick91) [inline tests/schema/test_generics.py:352]: this test might be useful 🤔

File: tests/test_printer.py [hunk 4]
```diff
@@ -32,22 +32,6 @@ class Query:
     assert print_schema(schema) == textwrap.dedent(expected_type).strip()
 
 
-# def test_recursive_type():
```
Comment (patrick91) [inline tests/test_printer.py:35]: same for this one :)

File: tests/schema/test_generics.py (hunk 3 above)
Comment (flpStrri) [reply tests/schema/test_generics.py:352]: 😆 Ok, I will restore it.

File: tests/test_printer.py (hunk 4 above)
Comment (flpStrri) [reply tests/test_printer.py:35]: 😆 Ok, I will restore it.

Comment (flpStrri) [reply tests/test_printer.py:35]: This test is broken right now. I will try to fix it but not sure if I will be able with my inexistent background using 🍓.

Comment (flpStrri) [reply tests/test_printer.py:35]: The test was created here: https://github.com/strawberry-graphql/strawberry/commit/d37cdac746cb000e43e19d82d42e2aa8d9a58b61
Never ran. :D

Comment (patrick91) [reply tests/test_printer.py:35]: let's delete it then ;)

File: tests/schema/test_generics.py (hunk 3 above)
Comment (flpStrri) [reply tests/schema/test_generics.py:352]: This was commented at that same point. I will try to make it pass now.

Comment (flpStrri) [reply tests/schema/test_generics.py:352]: I managed to make it pass but... why this order matters?
Captura de Tela 2020-09-04 às 13 03 36

Comment (patrick91) [reply tests/schema/test_generics.py:352]: good question, it shouldn’t matter! I’ll try to address this in another PR :)

Comment (flpStrri) [reply tests/schema/test_generics.py:352]: I think it is a good idea to open an issue about it, right?

**Review by patrick91 — APPROVED** [review]
Thanks so much for this! We managed to find a new bug too! 😊

I rebased and ignored some files for the time being as we are waiting for a new version of eradicate to get released, but we can still use this PR, we won't have many type: ignore comments anyway 😊

**Pull request merged**